### PR TITLE
test: Add taffy integration test

### PR DESCRIPTION
**Pull request by jvatic**
refs #942

### CODE REVIEW

File: test/test_taffy_deploy.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		},
+	})
+	t.Assert(err, c.IsNil)
+	attachClient := cluster.NewAttachClient(rwc)
+	var outBuf bytes.Buffer
+	exit, err := attachClient.Receive(&outBuf, &outBuf)
+	t.Log(outBuf.String())
+	outBuf.Reset()
+	t.Assert(exit, c.Equals, 0)
+	t.Assert(err, c.IsNil)
```
Comment (lmars) [inline test/test_taffy_deploy.go]: Could use `pkg/exec.Cmd`, unless there is a specific reason to go via the controller?

File: test/test_taffy_deploy.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	release, err := client.GetAppRelease(app.ID)
+	t.Assert(err, c.IsNil)
+
+	release = &ct.Release{
+		ArtifactID: artifact.ID,
+		Env:        release.Env,
+		Processes:  release.Processes,
+	}
+	t.Assert(client.CreateRelease(release), c.IsNil)
+	t.Assert(client.SetAppRelease(app.ID, release.ID), c.IsNil)
```
Comment (lmars) [inline test/test_taffy_deploy.go]: What is the purpose of changing the release to use the `test-apps` image? I think a comment would be good.

File: test/test_taffy_deploy.go (hunk 1 above)
Comment (titanous) [inline test/test_taffy_deploy.go]: This test emulates dashboard deploys (there should be a comment), hence the use of the controller.

File: test/test_taffy_deploy.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+			"app":        app.ID,
+		},
+	})
+	t.Assert(err, c.IsNil)
+	attachClient := cluster.NewAttachClient(rwc)
+	var outBuf bytes.Buffer
+	exit, err := attachClient.Receive(&outBuf, &outBuf)
+	t.Log(outBuf.String())
+	outBuf.Reset()
+	t.Assert(exit, c.Equals, 0)
```
Comment (lmars) [inline test/test_taffy_deploy.go]: Should we also test the actual outcome of running taffy? i.e. that the app got a new release?

**Comment by lmars** [comment]
It may be worth adding helper methods to make the test more readable (e.g. `deployWithTaffy(app, url)`)

**Comment by archSeer** [comment]
@jvatic @titanous Pushed a fix to the same branch.

**Comment by titanous** [comment]
@jvatic please rebase this and address the comments.

File: test/test_taffy_deploy.go (hunk 2 above)
Comment (jvatic) [inline test/test_taffy_deploy.go]: This is emulating [the dashboard](https://github.com/flynn/flynn/blob/ab3478079d1ae3418f3c97a5581e43bed0076d09/dashboard/app/lib/javascripts/dashboard/stores/app.js#L422-L449). As [releases no longer require an artifact](https://github.com/flynn/flynn/commit/6be3d4376bb4d2405b622f9b7f2444cd4c0743de), this step can be removed from both the test and dashboard app.

**Comment by jvatic** [comment]
@lmars @titanous comments addressed, rebased on latest master

(see 03af0bc42d8ff2ca48f9caccc0ce89b554849e21 and 488860aed9149aff8f3b245115904ea9545794b9 for changes squashed into original commit)

File: test/test_taffy_deploy.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	c "github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-check"
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/pkg/cluster"
+)
+
+type TaffyDeploySuite struct {
+	Helper
+}
+
+var _ = c.Suite(&TaffyDeploySuite{})
```
Comment (titanous) [inline test/test_taffy_deploy.go]: I think this can be marked as a `ConcurrentSuite`?

**Comment by titanous** [comment]
LGTM pending squash

**Pull request merged**